wiki/docs/developer-resources/functional-spec.md

```diff
@@ -0,0 +1,218 @@
+|**Background directory**|No|String|None|The background directory from which to retrieve backgrounds (optional, only applies if 'rotate background' is on)|
+|**Local backgrounds**|No|Boolean|True|Use local backgrounds (boolean, default on). If off, retrieves backgrounds from unsplash.it|
+|**Maximum images**|No|Whole number (>1)|10|Maximum number of images to store, only applies is "use local backgrounds" is off. (optional, defaults to 10, should be set at least 1)|
+
+#### View-specific settings
+Each view can have a tab with view specific settings which is only required if the view actually has settings to configure. As an example, the intent view today has no settings, but the clock view can have settings.
+
+Here are some examples of settings per view. This is not exhaustive.
+
+##### Clock
+
+|Setting name|Required yes/no|Type|Default value|Notes|
+|---|---|---|---|---|
+|Clock type|Yes|Enum (big, small, animated)|big|
+|Show weather|Yes|Boolean|True|
+|Show date|Yes|Boolean|True|
+|Animation speed (s)|No|Whole number|30|Applies only if clock type is set to animated|
+
+![Clock settings](functional-spec-images/clock-settings.png)
+
+##### Weather
+
+|Setting name|Required yes/no|Type|Default value|Notes|
+|---|---|---|---|---|
+|Weather entity|Yes|HA entity name in `weather` domain|If only one entity in `weather` domain then that is the default, otherwise None|Weather entity to use to show current weather and forecast with. This should be a dropdown of HA entities in the `weather` domain.|
+|Show forecast|Yes|Boolean|True|
+
+##### Adding a view
+WHen a user selects the add view button in the tabstrip containing the settings pages, the add view page opens:
+![Add view](functional-spec-images/add-view.png)
+
+Here, the user can add a new view by entering a unique view name, it's view type (any of the built-in view types or custom) and the access sentences. If any of these sentences are uttered the view should open.
+If the user sets the view type to custom the view code box shows in which the user can type/paste YAML code for the view.
+Add and cancel buttons are provided. If the user selects 'add view' the view is added to the tabstrip and if it is a built-in view its configuration settings are shown on that page.
+
+##### Access control settings
+The access control settings page allows the user to determine which views are accessible to which satellites. By default all views are available to all satellites. By disabling the correct checkbox a view can be made inaccessible to a satellite, which means that that satellite cannot access the view even if the correct view access sentence is uttered. The default view is always accessible by all satellites.
+
+![Access control](functional-spec-images/access-control.png)
+
+### Panel behavior
+1. By default the `default view` is shown on the satellite's display.
+1. The user can activate other views by uttering access sentences. The view is shown on the satellite's device that the user is using.
```

Comment:
For future revisions we could implement something that allows touch control too, I know it's mentioned above but I'm thinking views that allow full touch control (could be an iframe for Music Assistant's UI for example). But not right now, let's get the basics working well first.